handler/EventHandler.go

```diff
@@ -0,0 +1,225 @@
+package handler
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/coocood/freecache"
+)
+
+//struct for eventhandler which handles the instance and the start of the api
+type EventReport struct {
+	InstanceID string `avro:"instance_id"`
+	EventType  string `avro:"event_type"`
+	EventMsg   string `avro:"event_message"`
+}
+
+type config struct {
+	ImportAPIURL string
+	AuthToken    string
+}
+
+type instance struct {
+	InstanceID                string   `json:"instance_id"`
+	NumberOfObservations      int64    `json:"total_observations"`
+	TotalInsertedObservations int64    `json:"total_inserted_observations,omitempty"`
+	State                     string   `json:"state"`
+	Events                    *[]event `json:"events, omitempty"`
+}
+
+type event struct {
+	Type          string `json:"type"`
+	Message       string `json:"message"`
+	MessageOffset string `json:"messageOffset"`
+}
+
+var failed = "failed"
+
+//method to set an event not really needed but testing made things easier
+func (e *EventReport) EventSetter(InstanceID string, EventType string, EventMessage string) {
```

Comment:
Is there ever a case where you could construct an EventReporter without setting all of these fields at the same time? If not change to NewEventReport(....)